internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+	literalExpr.Value = msg.GetLiteral().GetValue()
+	return literalExpr
+}
+
+// ConvertMessageToCommandConstantBooleanExpr converts a message.Expr to a command.ConstantBooleanExpr
+func ConvertMessageToCommandConstantBooleanExpr(msg *Expr) *command.ConstantBooleanExpr {
+	constantBooleanExpr := &command.ConstantBooleanExpr{}
+	constantBooleanExpr.Value = msg.GetConstant().GetValue()
+	return constantBooleanExpr
+}
+
+// ConvertMessageToCommandUnaryExpr converts a message.Expr to a command.UnaryExpr
+func ConvertMessageToCommandUnaryExpr(msg *Expr) *command.UnaryExpr {
+	unaryExpr := &command.UnaryExpr{}
+	unaryExpr.Operator = msg.GetUnary().GetOperator()
+	unaryExpr.Value = ConvertMessageToCommandExpr(msg.GetUnary().GetValue())
+	return unaryExpr
+}
+
+// ConvertMessageToCommandBinaryExpr converts a message.Expr to a command.BinaryExpr
+func ConvertMessageToCommandBinaryExpr(msg *Expr) *command.BinaryExpr {
+	binaryExpr := &command.BinaryExpr{}
+	binaryExpr.Operator = msg.GetBinary().GetOperator()
+	binaryExpr.Left = ConvertMessageToCommandExpr(msg.GetBinary().GetLeft())
+	binaryExpr.Right = ConvertMessageToCommandExpr(msg.GetBinary().GetRight())
+	return binaryExpr
+}
+
+// ConvertMessageToCommandExprSlice converts a []*message.Expr to []command.Expr
+func ConvertMessageToCommandExprSlice(msg []*Expr) []command.Expr {
+	msgExprSlice := []command.Expr{}
+	for i := range msg {
+		msgExprSlice = append(msgExprSlice, ConvertMessageToCommandExpr(msg[i]))
+	}
+	return msgExprSlice
+}
+
+// ConvertMessageToCommandFunctionExpr converts a message.Expr to a command.FunctionExpr
+func ConvertMessageToCommandFunctionExpr(msg *Expr) *command.FunctionExpr {
+	functionExpr := &command.FunctionExpr{}
+	functionExpr.Name = msg.GetFunc().GetName()
+	functionExpr.Distinct = msg.GetFunc().GetDistinct()
+	functionExpr.Args = ConvertMessageToCommandExprSlice(msg.GetFunc().GetArgs())
+	return functionExpr
+}
+
+// ConvertMessageToCommandEqualityExpr converts a message.Expr to a command.EqualityExpr
+func ConvertMessageToCommandEqualityExpr(msg *Expr) *command.EqualityExpr {
+	equalityExpr := &command.EqualityExpr{}
+	equalityExpr.Left = ConvertMessageToCommandExpr(msg.GetEquality().GetLeft())
+	equalityExpr.Right = ConvertMessageToCommandExpr(msg.GetEquality().GetRight())
+	equalityExpr.Invert = msg.GetEquality().Invert
+	return equalityExpr
+}
+
+// ConvertMessageToCommandRangeExpr converts a message.Expr to a command.RangeExpr
+func ConvertMessageToCommandRangeExpr(msg *Expr) *command.RangeExpr {
+	rangeExpr := &command.RangeExpr{}
+	rangeExpr.Needle = ConvertMessageToCommandExpr(msg.GetRange().GetNeedle())
+	rangeExpr.Lo = ConvertMessageToCommandExpr(msg.GetRange().GetLo())
+	rangeExpr.Hi = ConvertMessageToCommandExpr(msg.GetRange().GetHi())
+	return rangeExpr
+}
+
+// ConvertMessageToCommandExpr converts a message.Expr to a command.Expr
+func ConvertMessageToCommandExpr(msg *Expr) command.Expr {
+	if msg == nil {
+		return nil
+	}
```

Comment:
Why is there a nil check here, but not in the other functions
?

Reply:
Will add them.